[4.0] Move Clear Expired Cache menu item to the toolbar

brianteeman opened the pull request:
"Clear Expired Cache" is no different to other purge operations such as "Purge Unpublished: Links" in com_redirect and "Clear Index" in com_finder so there is no need for it to have its own menu item and it should simply be a button on the toolbar for "Clear Cache".

I think there is a small B/C issue because someone could have created a custom admin menu for "Clear Expired Cache" so I am only making this for Joomla 4. If that is wrong it can easily be backported to J3

This PR 

- removes the menu link to "Clear Expired Cache"
- removes the purge view
- adds the "Clear Expired Cache" button to the "Clear Cache" toolbar
- adds the warning message to the button (as seen in  "Clear Index" in com_finder)
- Clean up language strings
- remove from sidebar

brianteeman committed:
remove clear expired cache from menu presets

brianteeman committed:
remove clear expired cache from menu presets

brianteeman committed:
delete files

brianteeman committed:
delete files

brianteeman committed:
remove sidebar

brianteeman committed:
remove sidebar

brianteeman committed:
add button to toolbar

brianteeman committed:
add button to toolbar

brianteeman committed:
remove sidebar language string

brianteeman committed:
remove sidebar language string

brianteeman committed:
change redirect

brianteeman committed:
change redirect

brianteeman committed:
Add warning when you press the button

brianteeman committed:
Add warning when you press the button

brianteeman committed:
update warning text

brianteeman committed:
update warning text

brianteeman committed:
Clean up language strings

brianteeman committed:
Clean up language strings

brianteeman committed:
codestyle

brianteeman committed:
codestyle

ghost commented:
I have tested this item :white_check_mark: successfully on e27e3dedb7a3101906bdf7ae7a844cb5311a5610<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17650">issues.joomla.org/tracker/joomla-cms/17650</a>.</sub>

brianteeman committed:
forgot to saVE

brianteeman committed:
forgot to saVE

ciar4n committed:
Last of SCSS variables for now (#434)

* Last of SCSS variables

* Build

* Happy hound

* Happier hound

* Tree fix

laoneo committed:
Merge remote-tracking branch 'remotes/upstream/4.0-dev' into dev

laoneo committed:
Adapter is responsible for filename normalization (#446)

* Adapter is responsible for filename normalization

* Update tests and default some setting values

* Better English

* Revert to old file name normalization

brianteeman committed:
MIME type (#455)

It is usual to write MIME and not Mime as it is an acronym

brianteeman committed:
Dimensions (#456)

It is more normal to write Dimensions and not dimension

ciar4n committed:
Move modal title to header (#461)

* Move image title to modal header

* Hound

* Further amendments

* Hound

* Adjust image top margin

ciar4n committed:
Change grid size icons (#460)

laoneo committed:
Merge remote-tracking branch 'remotes/upstream/4.0-dev' into dev

laoneo committed:
Merge remote-tracking branch 'remotes/upstream/4.0-dev' into dev

laoneo committed:
Allow spaces for file names on upload (#448)

* Remove alphanum restrictions for local filenames

* Basename strips certain characters

* Revert "Basename strips certain characters"

This reverts commit cab2f5c

laoneo committed:
Don't strip out special characters for display (#463)

laoneo committed:
Dowenload file action (#464)

* Add content parameter

* Download file

* Fix base 64 decoding

* Invalid file change

* Support videos and files

laoneo committed:
Add a warning message to the queue (#465)

* Add a warning message to the queue

* Use constant

ciar4n committed:
[4.0] Global config media query fix (#18658)

* Config media query fix

* Consider sidebar open

wilsonge committed:
Add meta.js to stop codemirror error. Fixes #18604

bembelimen committed:
Fix wrong Model class name call (#18158)

wilsonge committed:
Add meta.js to addons.js for parity with Joomla 3. Better fix for #18604

Commit:
Fix class name reference (Fix #18670)

ciar4n committed:
[4.0] Some modal fixes (#18665)

* Some modal fixes

* Remove inline style

infograf768 committed:
[4.0] Getting rid of tips for admin modules on the model of site modules (#18646)

brianteeman committed:
[4.0] Remove new Field group description (#18638)

Useless description for the field group title field should have been removed before

Either it was missed or it came back during a merge from staging

laoneo committed:
Small cleanup in admin (#18672)

wilsonge committed:
Fix refreshing help sites. Fixes #18594

wilsonge committed:
Fix JS Error for multiple select elements. Fixes #18645

wilsonge committed:
Fix newsfeed creation. Fixes #18680

wilsonge committed:
Fix Related items module. Fixes #18653

laoneo committed:
Observers doesn't exist anymore (#18678)

wilsonge committed:
Remove redirect that caused race condition

wilsonge committed:
Fix batch tagging

wilsonge committed:
Fix frontend module editing. Fixes #18691

wilsonge committed:
The user id shouldn't be cleaned from the session. Fixes #18627

wilsonge committed:
Make file size an empty string

wilsonge committed:
Make directory file size an empty string.

Fixes https://github.com/joomla/joomla-cms/issues/18616

wilsonge committed:
Fix uninstalling a component

joomdonation committed:
Fix typo in CalendarField (#18704)

laoneo committed:
Remove the update file of an unsupported database (#18675)

laoneo committed:
[4.0] Drop the not needed params column (#18676)

* Drop the not needed params column

* Remove the columns in the install files as well

laoneo committed:
Merge remote-tracking branch 'remotes/upstream/4.0-dev' into dev

Commit:
update tinymce to 4.7.2 (#18708)

laoneo committed:
Add a space before the copyright sign (#18707)

wilsonge committed:
Tag 4.0.0 Alpha 2

wilsonge committed:
Reset for dev

wilsonge committed:
Fix media form field for nested websites

wilsonge committed:
Fix editing template settings in the frontend

wilsonge committed:
Fix editing site settings in the frontend. Fixes #18726

wilsonge committed:
Fix forms not being loaded

wilsonge committed:
Fix forms not being loaded

laoneo committed:
Merge remote-tracking branch 'remotes/upstream/4.0-dev' into dev

# Conflicts:
#	administrator/components/com_media/Controller/ApiController.php
#	administrator/components/com_media/Model/ApiModel.php
#	plugins/filesystem/local/Adapter/LocalAdapter.php

C-Lodder committed:
Fix script URL (#18722)

C-Lodder committed:
Fix script URL (#18722)

Commit:
wc-fix (#18706)

Commit:
wc-fix (#18706)

C-Lodder committed:
[4.0] Update mediaelement and diff libraries (#18709)

* Update mediaelement and diff libraries

* update grunt-settings.yaml

C-Lodder committed:
[4.0] Update mediaelement and diff libraries (#18709)

* Update mediaelement and diff libraries

* update grunt-settings.yaml

wilsonge committed:
Fix not being able to create a contact. Fixes #18729

wilsonge committed:
Fix not being able to create a contact. Fixes #18729

wilsonge committed:
Fix saving a custom field. Fixes #18711 . Thanks @phalouvas

wilsonge committed:
Fix saving a custom field. Fixes #18711 . Thanks @phalouvas

laoneo committed:
[4.0] Reverts #18676 and initializes the params (#18753)

* Revert "[4.0] Drop the not needed params column (#18676)"

* Revert "[4.0] Drop the not needed params column (#18676)"

* Default the params field with an empty string

* Params should be valid json

laoneo committed:
[4.0] Reverts #18676 and initializes the params (#18753)

* Revert "[4.0] Drop the not needed params column (#18676)"

* Revert "[4.0] Drop the not needed params column (#18676)"

* Default the params field with an empty string

* Params should be valid json

C-Lodder committed:
Fix tab and slide pagebreak types (#18754)

C-Lodder committed:
Fix tab and slide pagebreak types (#18754)

C-Lodder committed:
Namespace Atum (#18752)

C-Lodder committed:
Namespace Atum (#18752)

wilsonge committed:
Fix com_finder giving javascript error. Fixes #18728 (#18737)

wilsonge committed:
Fix com_finder giving javascript error. Fixes #18728 (#18737)

brianteeman committed:
[4.0] Cassiopeia template previews (#18746)

Replace the protostar previews with the previews for cassiopeia template used in the template manager

brianteeman committed:
[4.0] Cassiopeia template previews (#18746)

Replace the protostar previews with the previews for cassiopeia template used in the template manager

C-Lodder committed:
Only load dragula is items exist (#18749)

C-Lodder committed:
Only load dragula is items exist (#18749)

ciar4n committed:
Style pagebreak links (#18759)

ciar4n committed:
Style pagebreak links (#18759)

C-Lodder committed:
Fix JText error on Archive view (#18780)

C-Lodder committed:
Fix JText error on Archive view (#18780)

C-Lodder committed:
Temporary fix for tags input width (#18784)

C-Lodder committed:
Temporary fix for tags input width (#18784)

Commit:
unify the error page for all cases (#18777)

Commit:
unify the error page for all cases (#18777)

C-Lodder committed:
Use namespaced classes for Cassiopeia (#18776)

C-Lodder committed:
Use namespaced classes for Cassiopeia (#18776)

C-Lodder committed:
[4.0] Fix dropdown menu alignment (#18782)

* Fix dropdown menu alignment

* indentation

C-Lodder committed:
[4.0] Fix dropdown menu alignment (#18782)

* Fix dropdown menu alignment

* indentation

mvanvu committed:
Fix JS searchtools wrong syntax. (#18769)

mvanvu committed:
Fix JS searchtools wrong syntax. (#18769)

C-Lodder committed:
[4.0] Styling for FE editing buttons (#18775)

* Styling for FE editing buttons

* Update edit.php

C-Lodder committed:
[4.0] Styling for FE editing buttons (#18775)

* Styling for FE editing buttons

* Update edit.php

C-Lodder committed:
Fix console log error from passwordview.js (#18755)

C-Lodder committed:
Fix console log error from passwordview.js (#18755)

richard67 committed:
Correct wrong tag "base" to "head" in templates/system/incompatible.html (#18796)

This error came from PR #18777

richard67 committed:
Correct wrong tag "base" to "head" in templates/system/incompatible.html (#18796)

This error came from PR #18777

zero-24 committed:
drone php minimum checks (#18804)

zero-24 committed:
drone php minimum checks (#18804)

C-Lodder committed:
Fix tooltips on frontend editing (#18786)

C-Lodder committed:
Fix tooltips on frontend editing (#18786)

laoneo committed:
Detects correctly if the files exists in API model (#471)

laoneo committed:
Check if the path is properly formated (#472)

* Check if the path is properly formated

* Use the right variable

zero-24 committed:
checkAccess thanks @laoneo (#18735)

zero-24 committed:
add the missing files to the ignore array (#18801)

wilsonge committed:
Add isSupported methods to the crypto classes

asika32764 committed:
Fix tests and use phpunit 6 (#18870)

wilsonge committed:
Start to fix tests

mbabker committed:
Fix undefined var

keradus committed:
DX: Further upgrade to PHPUnit 6 (#18874)

C-Lodder committed:
cpanel styling updates (#18827)

wilsonge committed:
Fix bootstrap tests

mbabker committed:
Remove polyfill libraries from core dependencies (#18875)

wilsonge committed:
Correct function call

wilsonge committed:
Fix media manager failing test

wilsonge committed:
Fix unit test after 1d445bd64908759176fa646b3b81b1ef7157b85e

wilsonge committed:
Refactor uses of getMockForAbstractClass to use PHPUnit mock builder

wilsonge committed:
Fix uses of getMock in Unit Tests

wilsonge committed:
Fix header in file

asika32764 committed:
[4.0] A more flexible ActionButton class (#18501)

brianteeman committed:
[4.0] Missing strings in admin menu (#18862)

Revert three strings

PR for #18858

laoneo committed:
Load the template script in defer mode (#18873)

mvanvu committed:
[4.0] searchtools should also clear search field (#18810)

* [4.0] searchtools should also clear search field

* Add spaces for condition

* [4.0] Update searctools.min.js

* Update searchtools.min.js use UglifyJS-online

ciar4n committed:
Fix debug console (#18837)

wilsonge committed:
Use correct mocking method

wilsonge committed:
Fix bootstrap assertation

wilsonge committed:
Fix bootstrap assertation

wilsonge committed:
Fix context string expected

wilsonge committed:
The cli script got deleted. Use one that does exist

wilsonge committed:
Fix code style

wilsonge committed:
Fix more codestyle

wilsonge committed:
Cli App Injection (#18878)

* Fix tests for CLI App input injection

* Remove sanity check

* ...and the new line

wilsonge committed:
Fix final test

C-Lodder committed:
[4.0] Fix drone errors? (#18887)

* Fix drone errors

* another fix test

ciar4n committed:
Modifier Classes - Featured Articles (#18883)

wilsonge committed:
Fix saving articles without tag data

C-Lodder committed:
Update tinyMCE and Codemirror (#18885)

Fedik committed:
[4.0]  Custom events: support in showon, templates, switcher, color field, tinymce (#18864)

* custom events: showon support `joomla:updated`

* custom events: template Cassiopeia support `joomla:updated`

* custom events: template Atum support `joomla:updated`

* custom events: switcher.js support `joomla:updated`

* custom events: Color field support `joomla:updated`

* custom events: tinymce init, support `joomla:updated`

* restore changes after conflict

wilsonge committed:
Add libsodium param

wilsonge committed:
Add libsodium into build

wilsonge committed:
Install the library first

wilsonge committed:
Fix URL and add source note

wilsonge committed:
Try installing a different way

wilsonge committed:
Update unit-tests.sh

wilsonge committed:
Try again with libsodium recognition

wilsonge committed:
Try LD Library path?

wilsonge committed:
Document progress

Commit:
Properly register ContentHelperRoute in modules (#18953)

infograf768 committed:
[4.0] Firefox: Correcting email field considered as a username field (#18911)

* [4.0] Correcting password autocomplete and emailfield considered as a
password field

* removing extra spaces

* Taking off code implementing autocomplete to off

joomdonation committed:
[4.0] Fix saving user profile error (#18705)

* Fix saving user profile error

* Remove un-used code (it is called later)

* Missing $db->quote

Commit:
[4.0] Clean up inline js #1 (#18944)

* com_categories

* Update admin-category-modal.min.js

* oops, this is totally useless

* all the rest views that contain this deprecated code

Commit:
com_config (#18951)

laoneo committed:
[4.0] Fix the sample data blog plugin (#18888)

* Fix to load the models and tables correct in the sample data blog plugin

* Use the correct table

* Fix positions

* Change links to default MVC

laoneo committed:
Check if the path is within the root folder (#467)

C-Lodder committed:
mod_multilangstatus improvements (#18947)

C-Lodder committed:
mod_popular improvements (#18938)

Commit:
Add support for predefined colors (Swatches) to Color field advanced layout (#18933)

C-Lodder committed:
mod_latest improvements (#18940)

C-Lodder committed:
[4.0] [mod_stats_admin] Improvements (#18950)

* mod_stats_admin improvements

* oops

C-Lodder committed:
[4.0] [mod_feed] Improvements (#18939)

* mod_feed improvements

* Update FeedHelper.php

C-Lodder committed:
mod_logged improvements (#18941)

laoneo committed:
Deprecate CMSApplication::getInstance (#18899)

laoneo committed:
Installation will use the global container (#18896)

wilsonge committed:
Travis Work

wilsonge committed:
Fix path

wilsonge committed:
More fixes to travis

C-Lodder committed:
mod_sampledata improvements (#18949)

C-Lodder committed:
mod_login improvements (#18942)

C-Lodder committed:
mod_menu improvements (#18946)

C-Lodder committed:
mod_quickicon improvements (#18948)

C-Lodder committed:
mod_custom improvements (#18943)

C-Lodder committed:
Fix stacking on close button

C-Lodder committed:
Add tab-pane styling to SCSS

C-Lodder committed:
remove inline CSS

C-Lodder committed:
Update autoprefixer (#18963)

rdeutz committed:
cs fix

ciar4n committed:
Recompile scss

wilsonge committed:
Merge pull request #478 from joomla-projects/compile-scss

Recompile

ciar4n committed:
[4.0] Move pageclass to body tag (#18967)

* Move pageclass to body tag

* thx charlie

C-Lodder committed:
[4.0] Replace BS alerts with CE alerts (#18872)

joomdonation committed:
Correct default value for $prefix parameter (#19004)

laoneo committed:
Use the namespace for the fieldgroups field in Contacts (#19008)

laoneo committed:
Merge remote-tracking branch 'remotes/upstream/4.0-dev' into dev

# Conflicts:
#	administrator/templates/atum/index.php
#	grunt-settings.yaml
#	templates/cassiopeia/css/template.css.map
#	templates/cassiopeia/css/template.min.css
#	tests/unit/suites/plugins/filesystem/local/LocalAdapterTest.php

C-Lodder committed:
Change some menu icons (#19000)

laoneo committed:
Make crop controls work (#480)

* Make crop controls work

* Simplify

laoneo committed:
Moved resize to canvas and removed some redundant code (#481)

laoneo committed:
Rotate the image more accurately and add an input box (#482)

mbabker committed:
Bump authentication and sodium_compat packages to improve Argon2i checks

mbabker committed:
Pull fix

brianteeman committed:
[4.0] Remove _QQ_ (#19030)

* [4.0] Remove _QQ_

Replaces all instances of _QQ_ in the language files with \" as per @nikosdion comments

I am unsure that I need to also change
```
		$strings  = @parse_ini_string($contents);
```
to set it to use $scanner_mode = INI_SCANNER_RAW

Please advise

* change to raw

stutteringp0et committed:
escaped some classes (#19019)

unescaped classes causing issues saving default values in fields - maybe other stuff too.

infograf768 committed:
Missing "info" lang string (#19035)

C-Lodder committed:
mod_toolbar improvements (#18988)

C-Lodder committed:
mod_title improvements (#18987)

C-Lodder committed:
[4.0] [mod_status] Improvements (#18986)

* mod_status improvements

* Fix

* add missing semicolon

C-Lodder committed:
[4.0] Rewrite indexer in vanilla Javascript (#18959)

* Rewrite indexer in vanilla Javascript

* revert to jQuery Ajax

* Fix error + call minified file

C-Lodder committed:
update quickicon icons (#19037)

joomdonation committed:
Fix tinymce Illegal string offset error (#18996)

ciar4n committed:
Fix up plugin UI (#484)

wilsonge committed:
Comment out libsodium for now

brianteeman committed:
[4.0] Remove language options on non multilingual site (#18972)

infograf768 committed:
[4.0] Multilingual: Only display the articles field 'show_associations' when associations are set. (#19034)

* [4.0] Multilingual: Only display the articles field 'show_associations'
when associations are set.

* Simplifying. Correcting some comments.

* cs

C-Lodder committed:
[4.0] Fix missing filters (com_menus & com_associations) (#19044)

* Fix filter for com_menus

* more fixes

cheesegrits committed:
Modification to Joomla.loadOptions() so it actually merges option obj… (#19052)

* Modification to Joomla.loadOptions() so it actually merges option objects.

* Tweak to Joomla.extend() to act like jQuery.extend() and treat a null destination as an empty object, rather than erroring out.

* Code style

* Code style (tabs)

* Code style (tabs).  Again.  Grrr.

mbabker committed:
Use long array syntax to avoid parse errors on PHP 5.3

C-Lodder committed:
Update dependencies (#19090)

C-Lodder committed:
[4.0] Start moving JS outside of template (#19086)

* Start moving JS outside of template

* move sidebar to mod_menu

* Move closest function back to files

laoneo committed:
Merge remote-tracking branch 'remotes/upstream/4.0-dev' into dev

laoneo committed:
CSS CS

laoneo committed:
Rebuild assets

laoneo committed:
CS Media Helper

laoneo committed:
CS Action plugins

laoneo committed:
CS Local Adapter

laoneo committed:
CS Media Helper

laoneo committed:
Language fixes

wilsonge committed:
Merge pull request #19095 from Digital-Peak/j4/media-manager

[4.0] Media Manager update

C-Lodder committed:
admin mod-login styling updates (#19087)

C-Lodder committed:
Fix badges on update quickicons (#19002)

brianteeman committed:
Merge remote-tracking branch 'brian/expired_cache' into expired_cache

# Conflicts:
#	administrator/components/com_cache/Controller/Controller.php
#	administrator/components/com_cache/View/Cache/Html.php
#	administrator/components/com_checkin/Controller/Controller.php

brianteeman commented:
one day I will learn how to resolve conflicts

brianteeman committed:
Merge remote-tracking branch 'brian/expired_cache' into expired_cache

# Conflicts:
#	administrator/components/com_cache/Controller/Controller.php
#	administrator/components/com_cache/View/Cache/Html.php
#	administrator/components/com_checkin/Controller/Controller.php (reverted from commit c2205cac6a21175b35439bd6a15527c74cdc91c3)
